application_manager/plugins/spark_mesos/plugin.py

```diff
@@ -0,0 +1,250 @@
+        print o.read()
+
+        plugin_log.log("%s | Stopping monitor" % (time.strftime("%H:%M:%S")))
+        monitor.stop_monitor(api.monitor_url, self.app_id)
+
+        plugin_log.log("%s | Stopping scaler" % (time.strftime("%H:%M:%S")))
+        scaler.stop_scaler(api.controller_url, self.app_id)
+
+        plugin_log.log("%s | Remove binaries" % (time.strftime("%H:%M:%S")))
+        conn.exec_command('rm -rf ~/exec_bin.*')
+        return None, None
+
+    def _get_ssh_connection(self, ip, username=None,
+                            password=None, key_path=None):
+        # Preparing SSH connection
+        conn = paramiko.SSHClient()
+        conn.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+        # Checking if the connection will be established using
+        # keys or passwords
+        if key_path != "" and key_path is not None:
+            keypair = paramiko.RSAKey.from_private_key_file(key_path)
+            conn.connect(hostname=ip, username=username, pkey=keypair)
+        else:
+            conn.connect(hostname=ip, username=username, password=password)
+
+        return conn
+
+    def _get_executors_ip(self, conn):
+        frameworks_call = ('curl http://' + self.get_frameworks_url
+                           + '/frameworks')
+
+        time.sleep(5)
+        stdin, stdout, sterr = conn.exec_command(frameworks_call)
+
+        try:
+            output = stdout.read()
+            mesos_resp = json.loads(output)
+        except Exception as e:
+            print e.message
+            mesos_resp = {}
+
+        executors_ips = []
+        framework = None
+        find_fw = False
+
+        # It must to ensure that the application was
+        # started before try to get the executors
+        while not find_fw:
+            for f in mesos_resp['frameworks']:
+                if f['name'] == self.app_id:
+                    framework = f
+                    find_fw = True
+                    break
+            if not find_fw:
+                plugin_log.log("%s | Trying to find framework" %
+                              (time.strftime("%H:%M:%S")))
+
+                stdin, stdout, sterr = conn.exec_command(frameworks_call)
+                mesos_resp = json.loads(stdout.read())
+
+            time.sleep(2)
+
+        # Look for app-id into the labels and
+        # get the framework that contains it
+        for t in framework['tasks']:
+            for s in t['statuses']:
+                for n in s['container_status']['network_infos']:
+                    for i in n['ip_addresses']:
+                        executors_ips.append(i['ip_address'])
+
+        plugin_log.log("%s | Executors IPs: %s"
+                        % (time.strftime("%H:%M:%S"), executors_ips))
+
+        plugin_log.log("%s | WebUI URL: %s"
+                        % (time.strftime("%H:%M:%S"), framework['webui_url']))
+
+        return executors_ips, framework['webui_url']
+
+    def _extract_vms_ids(self, output):
```

Comment:
I think this method should be moved to "utils/mesos.py"

Reply:
I agree.